### PR TITLE
javascript-light-bulb - Matthew Kweon

### PR DESCRIPTION
**Pull request by MattKweon**
Kapture 2022-09-06 at 16 38 31

### CODE REVIEW

File: javascript-light-bulb/main.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+var $container = document.querySelector('.container');
+var $lightSwitch = document.querySelector('.light-switch');
+var count = 0;
+function lightSwitchFlicks() {
+  count++;
+  if (count % 2 !== 0) {
```
Comment (codingcodymiller) [inline javascript-light-bulb/main.js]: To determine whether to turn the light on or off, you are making a counter, but the only thing you ever care about is if the counter's value was even or odd. This is a situation where you would be better off using a boolean to keep track of if the light is currently on.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by MattKweon** [comment]
Kapture 2022-09-07 at 15 31 14

File: javascript-light-bulb/main.js [hunk 2]
```diff
@@ -1,9 +1,8 @@
 var $container = document.querySelector('.container');
 var $lightSwitch = document.querySelector('.light-switch');
-var count = 0;
+
 function lightSwitchFlicks() {
-  count++;
-  if (count % 2 !== 0) {
+  if ($container.className === 'container off') {
```
Comment (codingcodymiller) [inline javascript-light-bulb/main.js]: Now your logic is dependent on the state of the DOM to know what to do, which is less than ideal. You should follow my initial suggestion of using a variable containing a boolean value to keep track of if the light is currently on or not, and use that variable's value to decide what action to take when the lightbulb is clicked.

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by MattKweon** [comment]
Kapture 2022-09-08 at 08 59 14

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**